addons/barcodes/static/src/js/barcode_events.js

```diff
@@ -0,0 +1,148 @@
+odoo.define('web.BarcodeEvents', function(require) {
+"use strict";
+
+var core = require('web.core');
+var mixins = core.mixins;
+
+var BarcodeEvents = core.Class.extend(mixins.PropertiesMixin, {
+    timeout: null,
+    key_pressed: {},
+    buffered_key_events: [],
+    // Regexp to match a barcode input and extract its payload
+    // Note: to build in init() if prefix/suffix can be configured
+    regexp: /(.{3,})[\n\r\t]+/,
+    // By knowing the terminal character we can interpret buffered keys
+    // as a barcode as soon as it's encountered (instead of waiting x ms)
+    suffix: /[\n\r\t]+/,
```

Comment:
What is the advantage of introducing this suffix? Is it just to avoid having to wait `max_time_between_keys_in_ms` one time at the end?